### ISSUE TITLE
Improve/replace srcd sql client

### ISSUE THREAD

**Issue by smacker** [src-d/engine#154]
I had expected to have the same experience as with standard mysql client. Few things I missed:

- Can't switch to display results per column (`\G` in the normal client)
- No pager
- When I use some special keys it messes with the output (starting removing lines and so on)

**Comment by carlosms** [comment]
Maybe we could detect if you have the mysql client installed and launch it, and keep the current code as a fallback. This way we don't add a hard requirement, but still improve the experience using the regular client when possible.

cc @campoy

**Comment by smacker** [comment]
We can also run `docker exec -it mysql-cli/image mysql`.

**Comment by smacker** [comment]
I started working on it. List of the issue that will be solved (requested by Santiago):
- https://github.com/src-d/engine/issues/278
- https://github.com/src-d/engine/issues/215
- https://github.com/src-d/engine/issues/122
- https://github.com/src-d/engine/issues/257
- https://github.com/src-d/engine/issues/271

**Comment by se7entyse7en** [comment]
Ok, I've just self-assigned #271, but I haven't actually started working on it. Given that all those issues don't seem to be urgent, maybe better to not start the issues that this one will fix?

**Comment by marnovo** [comment]
Giving the green light.

Backlog issue created at: https://github.com/src-d/backlog/issues/1387

From Slack:

> Santiago Mola [12:01 PM]
> can you look at this? https://github.com/src-d/engine/issues/154
> GitHub
> Improve/replace srcd sql client · Issue #154 · src-d/engine
> I had expected to have the same experience as with standard mysql client. Few things I missed: Can&#39;t switch to display results per column (\G in the normal client) No pager When I use some spec...
> 
> since I told them to take significant fixes in, they wanted to start with this, which should solve quite a few usability issues and bugs in one go
> 
> Marcelo Novaes [12:23 PM]
> Ok, so in practice this means killing a custom REPL and replacing with the mysql one OR working ours to be more akin to mysql one?
> 
> Santiago Mola [12:25 PM]
> Yes, it's replacing our current REPL with the official mysql client
> 
> Marcelo Novaes [12:25 PM]
> Haha could be written in plain English like that :joy:
> Slack couldn’t send this messageTry again
> | Cancel
> 
> Santiago Mola [12:26 PM]
> mysql official repl is not awesome but works and it's what DP team actually uses
> I'd say it's a good move until the day we decide we want to do our own super awesome REPL
> 
> Marcelo Novaes [12:30 PM]
> Sounds sensible, do we have an estimation? Can we get this working/integrated/tested before the release?
> 
> Santiago Mola [12:34 PM]
> there should be a PR for it on Monday, so yes
> Monday/Tuesday, given the OSD
> 
> Marcelo Novaes [12:34 PM]
> Great, I’ll paste the conversation there and give the green light.

**Comment by carlosms** [comment]
> Ok, I've just self-assigned #271, but I haven't actually started working on it. Given that all those issues don't seem to be urgent, maybe better to not start the issues that this one will fix?

I think #271 is not related to the mysql client change, it must have been a typo. @smacker?

**Comment by smacker** [comment]
true. Must be mixed up with some other number.
